app/src/resources/camera/usePreviewImage.ts

```diff
@@ -13,23 +16,44 @@ export interface UsePreviewImageResult {
 }
 
 export function usePreviewImage(
-  settings: CameraImageSettings
+  settings: CameraImageSettings,
+  runId: string | null
 ): UsePreviewImageResult {
   const [isLoading, setIsLoading] = useState(false)
-  const { data, refetch } = useCapturePreviewImage(settings, {
+
+  const previewQuery = useCapturePreviewImage(settings, {
+    enabled: runId == null,
     onSettled: () => {
       setIsLoading(false)
     },
     onError: error => {
       console.error('Failed to capture preview image', error)
     },
   })
+  console.log('🚀 ~ usePreviewImage ~ previewQuery:', previewQuery)
 
-  const imgPath = useMemo(() => {
-    if (data == null) return undefined
+  const previewWithRunQuery = useAddCapturePreviewImageToRun(
+    settings,
+    runId ?? '',
+    {
+      enabled: runId != null,
+      onSettled: () => {
+        setIsLoading(false)
+      },
+      onError: error => {
+        console.error(`Failed to capture preview image in run ${runId}`, error)
+      },
+    }
+  )
 
-    const blob = new Blob([data], { type: MIME_TYPES.IMAGE })
-    return URL.createObjectURL(blob)
+  const { data, refetch } = runId == null ? previewQuery : previewWithRunQuery
```

Comment:
```suggestion
  const { data, refetch } = runId == null ? previewQuery : previewWithRunQuery
```
This will end up looking something like:
`const takePhoto = runId == null ? capturePreviewImage : capturePreviewImageForRun`